nixos/lib/utils.nix

```diff
@@ -115,4 +115,8 @@ rec {
         config.flyingcircus.users.userData
       );
 
+  writePrettyJSON = name: x:
+    pkgs.runCommand name { preferLocalBuild = true; } ''
+      echo '${toJSON x}' | ${pkgs.jq}/bin/jq . > $out
+    '';
```

Comment:
Quoting correct? JSON strings may contain `'` IIRC

Reply:
I write raw json to a file now and pass that to jq. Avoids quoting problems and makes debugging jq errors easier.